src/posix/platform/radio_spinel.cpp

```diff
@@ -612,34 +612,37 @@ otError RadioSpinel::ParseRadioFrame(otRadioFrame &aFrame, const uint8_t *aBuffe
-        aBuffer, aLength,
-        SPINEL_DATATYPE_DATA_WLEN_S                              // Frame
-                        SPINEL_DATATYPE_INT8_S                   // RSSI
-                        SPINEL_DATATYPE_INT8_S                   // Noise Floor
-                        SPINEL_DATATYPE_UINT16_S                 // Flags
-                        SPINEL_DATATYPE_STRUCT_S(                // PHY-data
-                            SPINEL_DATATYPE_UINT8_S              // 802.15.4 channel
-                                        SPINEL_DATATYPE_UINT8_S  // 802.15.4 LQI
-                                        SPINEL_DATATYPE_UINT32_S // Timestamp (ms).
-                                        SPINEL_DATATYPE_UINT16_S // Timestamp (us).
-                            ) SPINEL_DATATYPE_STRUCT_S(          // Vendor-data
-                            SPINEL_DATATYPE_UINT_PACKED_S        // Receive error
-                            ),
-        aFrame.mPsdu, &size, &aFrame.mInfo.mRxInfo.mRssi, &noiseFloor, &flags, &aFrame.mChannel,
-        &aFrame.mInfo.mRxInfo.mLqi, &aFrame.mInfo.mRxInfo.mMsec, &aFrame.mInfo.mRxInfo.mUsec, &receiveError);
+    unpacked =
+        spinel_datatype_unpack_in_place(aBuffer, aLength,
+                                        SPINEL_DATATYPE_DATA_WLEN_S                              // Frame
+                                                        SPINEL_DATATYPE_INT8_S                   // RSSI
+                                                        SPINEL_DATATYPE_INT8_S                   // Noise Floor
+                                                        SPINEL_DATATYPE_UINT16_S                 // Flags
+                                                        SPINEL_DATATYPE_STRUCT_S(                // PHY-data
+                                                            SPINEL_DATATYPE_UINT8_S              // 802.15.4 channel
+                                                                        SPINEL_DATATYPE_UINT8_S  // 802.15.4 LQI
+                                                                        SPINEL_DATATYPE_UINT32_S // Timestamp (ms).
+                                                                        SPINEL_DATATYPE_UINT16_S // Timestamp (us).
+                                                            ) SPINEL_DATATYPE_STRUCT_S(          // Vendor-data
+                                                            SPINEL_DATATYPE_UINT_PACKED_S        // Receive error
+                                                            ),
+                                        aFrame.mPsdu, &size, &aFrame.mInfo.mRxInfo.mRssi, &noiseFloor, &flags,
+                                        &aFrame.mChannel, &aFrame.mInfo.mRxInfo.mLqi, &msec, &usec, &receiveError);
 
     VerifyOrExit(unpacked > 0, error = OT_ERROR_PARSE);
 
     if (receiveError == OT_ERROR_NONE)
     {
         aFrame.mLength = static_cast<uint8_t>(size);
 
+        aFrame.mInfo.mRxInfo.mTimestamp             = msec * 1000 + usec;
```

Comment:
The sniffer.py needs to be updated as well:
https://github.com/openthread/pyspinel/blob/master/sniffer.py#L195